src/Operation/Distinct.php

```diff
@@ -83,6 +90,10 @@ public function __construct(private string $databaseName, private string $collec
             throw InvalidArgumentException::expectedDocumentType('"collation" option', $this->options['collation']);
         }
 
+        if (isset($this->options['hint']) && ! is_string($this->options['hint']) && ! is_array($this->options['hint']) && ! is_object($this->options['hint'])) {
```

Comment:
The doc says `string|document`, the logical test should use `is_document`.

```suggestion
        if (isset($this->options['hint']) && ! is_string($this->options['hint']) && ! is_document($this->options['hint'])) {
```

Reply:
Good catch. I'll also make a separate ticket to migrate other `hint` option validation to use `is_document`, as it looks like we missed that when introducing PackedArray.

Reply:
Opened [PHPLIB-1587](https://jira.mongodb.org/browse/PHPLIB-1587) to track that.